### PR TITLE
Config handler increment tool version

### PR DESCRIPTION
**Pull request by eliasyishak**
This new feature will allow for a newer version of the message to increment the version number in the config file

### CODE REVIEW

File: lib/src/config_handler.dart [hunk 1]
```diff
@@ -102,6 +102,43 @@ class ConfigHandler {
...
+    // match the line for the current tool (ie. flutter-tools=2023-01-05,1)
+    final String configString = configFile.readAsStringSync();
+    final String pattern = '^($tool)=([0-9]{4}-[0-9]{2}-[0-9]{2}),([0-9]+)\$';
+
+    final RegExp regex = RegExp(pattern, multiLine: true);
+    final Iterable<RegExpMatch> matches = regex.allMatches(configString);
+
+    // TODO: need to determine what to do when there are two lines for the same tool
+    //  as outlined in the Dev Notes document; currently only assuming one line per tool
+    if (matches.length == 1) {
```
Comment (Jasguerrero) [inline lib/src/config_handler.dart]: Could we raise an exception as not implemented when `matches.length > 1` and a test for that?

Comment (eliasyishak) [reply lib/src/config_handler.dart]: Great idea, I actually should just implement the method that will get triggered when the matches aren't equal to 1 so I can get rid of this TODO

Comment (eliasyishak) [reply lib/src/config_handler.dart]: Just added the functionality to reset the configuration file if that conditional fails. Also added tests for it

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Pull request merged**